Ladybird setup fails due to missing `rustc`

Following the [instructions](https://github.com/nix-community/nix-environments/blob/8eb3d2d33a8195bfffef83787e0f5ca3fb04ac7f/envs/ladybird/README.md) for setting up Ladybird, I get the error:

```
CMake Error at Build/release/_deps/corrosion-src/cmake/FindRust.cmake:23 (message):
  `rustc` not found in PATH or `/home/<user>/.cargo/bin`.

  Hint: Check if `rustc` is in PATH or manually specify the location by
  setting `Rust_COMPILER` to the path to `rustc`.
Call Stack (most recent call first):
  Build/release/_deps/corrosion-src/cmake/FindRust.cmake:247 (_findrust_failed)
  Build/release/_deps/corrosion-src/cmake/Corrosion.cmake:63 (find_package)
  Build/release/_deps/corrosion-src/CMakeLists.txt:73 (include)
```

My build also fails (commit 502f261@ladybird). 

```
rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/rustc: error while loading shared libraries: libz.so.1: cannot open shared object file: No such file or directory
CMake Warning (dev) at Build/release/_deps/corrosion-src/cmake/FindRust.cmake:362 (message):
  Unexpected output from `rustc --version` for Toolchain
  `nightly-x86_64-unknown-linux-gnu`: ``.

  Ignoring this toolchain.
Call Stack (most recent call first):
  Build/release/_deps/corrosion-src/cmake/Corrosion.cmake:63 (find_package)
  Build/release/_deps/corrosion-src/CMakeLists.txt:73 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Rust Toolchain: stable-x86_64-unknown-linux-gnu
-- Rust Target: x86_64-unknown-linux-gnu
-- Determining required link libraries for target x86_64-unknown-linux-gnu
-- Required static libs for target x86_64-unknown-linux-gnu: gcc_s;util;rt;pthread;m;dl;c
-- Required link flags for target x86_64-unknown-linux-gnu:
-- Found Rust: /home/aaaaa/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/rustc (found version "1.94.0")
-- Using Corrosion as a subdirectory
-- Could NOT find cpptrace (missing: cpptrace_DIR)
-- Looking for backtrace
-- Looking for backtrace - found
-- backtrace facility detected in default set of libraries
-- Found Backtrace: /nix/store/j8irrc0mpx029dw0rmadsjylg7h31ync-glibc-2.42-51-dev/include
CMake Error at AK/CMakeLists.txt:75 (find_package):
  Could not find a package configuration file provided by "fmt" with any of
  the following names:

    fmtConfig.cmake
    fmt-config.cmake

  Add the installation prefix of "fmt" to CMAKE_PREFIX_PATH or set "fmt_DIR"
  to a directory containing one of the above files.  If "fmt" provides a
  separate development package or SDK, be sure it has been installed.


```

I also get the fmtConfig error on ladybird commit `b4185f0ecd8d85a28b6524325e7daec9f4fad271`

```
cmake -DBUILD_TESTING=True -GNinja -BBuild/test 
-- The C compiler identification is GNU 15.2.0
-- The CXX compiler identification is GNU 15.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/kbw2j1vag664b3sj3rjwz9v53cqx87sb-gcc-wrapper-15.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/kbw2j1vag664b3sj3rjwz9v53cqx87sb-gcc-wrapper-15.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CXX_COMPILER_SUPPORTS_BLOCKS
-- Performing Test CXX_COMPILER_SUPPORTS_BLOCKS - Failed
-- Performing Test CXX_COMPILER_SUPPORTS_OBJC_ARC
-- Performing Test CXX_COMPILER_SUPPORTS_OBJC_ARC - Failed
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test LAGOM_LINKER_SUPPORTS_LINKER:--gdb-index
-- Performing Test LAGOM_LINKER_SUPPORTS_LINKER:--gdb-index - Failed
-- Performing Test LAGOM_LINKER_SUPPORTS_LINKER:-Bsymbolic-non-weak-functions
-- Performing Test LAGOM_LINKER_SUPPORTS_LINKER:-Bsymbolic-non-weak-functions - Failed
-- Could NOT find cpptrace (missing: cpptrace_DIR)
-- Looking for backtrace
-- Looking for backtrace - found
-- backtrace facility detected in default set of libraries
-- Found Backtrace: /nix/store/j8irrc0mpx029dw0rmadsjylg7h31ync-glibc-2.42-51-dev/include
CMake Error at AK/CMakeLists.txt:75 (find_package):
  Could not find a package configuration file provided by "fmt" with any of
  the following names:

    fmtConfig.cmake
    fmt-config.cmake

  Add the installation prefix of "fmt" to CMAKE_PREFIX_PATH or set "fmt_DIR"
  to a directory containing one of the above files.  If "fmt" provides a
  separate development package or SDK, be sure it has been installed.


-- Configuring incomplete, errors occurred!
```

I also had problems with [this](https://github.com/nix-community/nix-environments/tree/master/envs/ladybird) tutorial. I get this error:
```
~ cmake -DENABLE_ADDRESS_SANITIZER=ON -DENABLE_UNDEFINED_SANITIZER=ON -GNinja -BBuild/san
-- The C compiler identification is GNU 15.2.0
-- The CXX compiler identification is GNU 15.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/hb2bs5fg5wkm04x565737qd5nh2hy5nk-gcc-wrapper-15.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/hb2bs5fg5wkm04x565737qd5nh2hy5nk-gcc-wrapper-15.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenSSL: /nix/store/bga5xf95jaypy385hvxm4h3yxl3m1566-openssl-3.6.1/lib/libcrypto.so (found version "3.6.1")
-- Performing Test CXX_COMPILER_SUPPORTS_BLOCKS
-- Performing Test CXX_COMPILER_SUPPORTS_BLOCKS - Failed
-- Performing Test CXX_COMPILER_SUPPORTS_OBJC_ARC
-- Performing Test CXX_COMPILER_SUPPORTS_OBJC_ARC - Failed
-- Performing Test LAGOM_LINKER_SUPPORTS_LINKER:--gdb-index
-- Performing Test LAGOM_LINKER_SUPPORTS_LINKER:--gdb-index - Failed
-- Performing Test LAGOM_LINKER_SUPPORTS_LINKER:-Bsymbolic-non-weak-functions
-- Performing Test LAGOM_LINKER_SUPPORTS_LINKER:-Bsymbolic-non-weak-functions - Failed
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
CMake Error at CMakeLists.txt:140 (find_package):
  Could not find a package configuration file provided by "mimalloc" with any
  of the following names:

    mimallocConfig.cmake
    mimalloc-config.cmake

  Add the installation prefix of "mimalloc" to CMAKE_PREFIX_PATH or set
  "mimalloc_DIR" to a directory containing one of the above files.  If
  "mimalloc" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
```

I think the shell is just missing mimalloc

--- a/envs/ladybird/README.md
+++ b/envs/ladybird/README.md
@@ -6,8 +6,13 @@
 
 Ladybird's build system uses vcpkg to vendor third-party dependencies, which proves undesirable to
 use with Nix for [several reasons](https://github.com/LadybirdBrowser/ladybird/issues/371).
-As a result, using `ladybird.sh` to compile and run Ladybird will fail. Therefore, it is necessary
+As a result, using `ladybird.py` to compile and run Ladybird will fail. Therefore, it is necessary
 to use system packages provided by this environment.
+
+This environment exports `ICU_ROOT` and sets `FONTCONFIG_FILE` (pointing to `dejavu_fonts` and
+`liberation_ttf`). Both are required: without `ICU_ROOT` cmake cannot locate ICU headers, and
+without the font configuration Ladybird crashes with `SIGILL` on startup because its font
+fallback lists require fonts not present in a default NixOS installation.
 
 Clone the repository & enter the shell:
 

--- a/envs/ladybird/shell.nix
+++ b/envs/ladybird/shell.nix
@@ -7,22 +7,25 @@ pkgs.mkShell {
   name = "ladybird-env";
 
   inputsFrom = [
-    (pkgs.ladybird.override (prev: {
-      skia = prev.skia.overrideAttrs (prev: {
-        gnFlags = prev.gnFlags ++ [
-          "extra_cflags_cc=[\"-frtti\"]"
-        ];
-      });
-    }))
+    pkgs.ladybird
   ];
 
   packages = with pkgs; [
     ccache
     clang-tools
     prettier
-    icu78
+    icu78.dev
+    mimalloc
+    dejavu_fonts
+    liberation_ttf
   ] ++ extraPkgs;
 
+  ICU_ROOT = "${pkgs.icu78.dev}";
+
+  FONTCONFIG_FILE = pkgs.makeFontsConf {
+    fontDirectories = [ pkgs.dejavu_fonts pkgs.liberation_ttf ];
+  };
+
   # https://github.com/NixOS/nixpkgs/blob/79a8a723b9/pkgs/by-name/la/ladybird/package.nix#L144-L147
-  NIX_LDFLAGS = "-lGL";
+  NIX_LDFLAGS = "-lGL -lfontconfig";
 }

